onnxruntime/contrib_ops/cpu/transformers/greedy_search_impl_gpt.h

```diff
@@ -0,0 +1,236 @@
+                                          feeds,
+                                          this->create_inputs_func_,
+                                          this->add_to_feeds_func_,
+                                          buffer);
+}
+
+template <typename T>
+Status GreedySearchGpt<T>::UpdateFeeds(
+    const std::vector<OrtValue>& last_outputs,
+    std::vector<OrtValue>& next_inputs,
+    int current_length,
+    OrtValue& position_ids,
+    gsl::span<const int32_t> next_tokens) {
+  gsl::span<const int32_t> place_holder;
+  return update_feeds_func_(this->temp_space_allocator_,
+                            this->cuda_stream_,
+                            last_outputs,
+                            next_inputs,
+                            current_length,
+                            position_ids,
+                            true,
+                            next_tokens,
+                            place_holder,
+                            this->parameters_->num_beams,
+                            gpt_subgraph_.GetFirstPastInputIndex(),
+                            gpt_subgraph_.GetFirstPresentOutputIndex());
+}
+
+template <typename T>
+Status GreedySearchGpt<T>::Execute(const FeedsFetchesManager& feeds_fetches_manager) {
+  auto status = Status::OK();
+  const GreedySearchParameters* parameters = this->parameters_;
+
+  // Allocate output tensors.
+  int64_t sequences_dims[] = {parameters->batch_size, parameters->max_length};
+  TensorShape sequences_shape(&sequences_dims[0], sizeof(sequences_dims) / sizeof(sequences_dims[0]));
+  Tensor* output_sequences = this->context_.Output(0, sequences_shape);
+
+  std::vector<OrtValue> feeds;
+  std::vector<OrtValue> fetches;
+
+  GreedySearchState<T> greedy_state;
+  greedy_state.Init(this->cpu_allocator_,
+                    this->temp_space_allocator_,
+                    static_cast<int>(parameters->BatchBeamSize()),
+                    static_cast<int>(parameters->vocab_size),
+                    static_cast<int>(parameters->sequence_length),
+                    parameters->max_length,
+                    this->IsCuda());
+
+  IAllocatorUniquePtr<char> buffer;
+  OrtValue expanded_input_ids_in_cpu;
+  ORT_RETURN_IF_ERROR(CreateInitialFeeds(greedy_state.sequence_lengths, expanded_input_ids_in_cpu, feeds, buffer));
+
+  init_greedy_state_func_(&greedy_state,
+                          greedy_state.sequence_lengths,
+                          parameters->batch_size,
+                          this->cuda_stream_);
+
+  gsl::span<const int32_t> input_ids = expanded_input_ids_in_cpu.Get<Tensor>().DataAsSpan<int32_t>();
+  greedy_state.SetSequence(input_ids,
+                           static_cast<size_t>(parameters->BatchBeamSize()),
+                           parameters->max_length,
+                           parameters->sequence_length);
+
+#ifdef DEBUG_GENERATION
+  const IConsoleDumper* dumper = this->GetConsoleDumper();
+  dumper->Print("input_ids", feeds[0]);
```

Comment:
Move the dump part inside while statement like beam_search_impl_gpt.h. Otherwise, only the first iteration has dump.